metrics/mean_iou/mean_iou.py

```diff
@@ -0,0 +1,237 @@
+# Copyright 2022 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Mean IoU (Intersection-over-Union) metric."""
+
+import numpy as np
+
+import datasets
+
+
+_DESCRIPTION = """
+IoU is the area of overlap between the predicted segmentation and the ground truth divided by the area of union
+between the predicted segmentation and the ground truth. For binary (two classes) or multi-class segmentation,
+the mean IoU of the image is calculated by taking the IoU of each class and averaging them.
+"""
+
+_KWARGS_DESCRIPTION = """
+Args:
+    predictions: Predicted segmentation maps.
+    references: Ground truth segmentation maps.
+    num_labels (int): Number of classes (categories).
+    ignore_index (int): Index that will be ignored in evaluation.
+    nan_to_num (int, optional): If specified, NaN values will be replaced by the numbers defined by the user. Default: None.
+    label_map (dict): Mapping old labels to new labels. Default: dict().
+    reduce_zero_label (bool): Whether ignore zero label. Default: False.
+Returns:
+    dict[str, float | ndarray]:
+            <aAcc> float: Overall accuracy on all images.
+            <Acc> ndarray: Per category accuracy, shape (num_labels, ).
+            <IoU> ndarray: Per category IoU, shape (num_labels, ).
+Examples:
+
+    >>> mean_iou_metric = datasets.load_metric("mean_iou")
+    >>> results = mean_iou_metric.compute(references=[0, 1], predictions=[0, 1])
+    >>> print(results)
+"""
+
+_CITATION = """\
+authors:
+  - name: "MMSegmentation Contributors"
+title: "OpenMMLab Semantic Segmentation Toolbox and Benchmark"
+date-released: 2020-07-10
+url: "https://github.com/open-mmlab/mmsegmentation"
+license: Apache-2.0
+"""
```

Comment:
```suggestion
_CITATION = """\
@software{MMSegmentation_Contributors_OpenMMLab_Semantic_Segmentation_2020,
author = {{MMSegmentation Contributors}},
license = {Apache-2.0},
month = {7},
title = {{OpenMMLab Semantic Segmentation Toolbox and Benchmark}},
url = {https://github.com/open-mmlab/mmsegmentation},
year = {2020}
}"""
```

Copied from the "Cite this repository" dialog.